### DISCUSSION TITLE
Local user for remote connections

### DISCUSSION THREAD

**Discussion by majkinetor** [1Remote/1Remote#392] [Ideas]
Think about a way to share a connection to a team, but allow somehow an option for each user to use its own username and password (or more likely domain account). 

Alternative usernames is not good for this scenario.

**Comment by VShawn** [comment]
> Alternative usernames is not good for this scenario.

Because the alternate user was not designed for this scenario.

> allow somehow an option for each user to use its own username 

I currently do not have a good design to use local user in Team shared connection. 

A few months ago, when designing the team share feature, I brainstormed about this feature: if we want to achieve it, then we must store a copy of the team connection ID locally and allow add a local user for each ID. 

Then, what should we do to the account stored locally when a team connection on MySQL is deleted by the DBA? We cannot delete it, but keeping it seems pointless since the connect on MySQL has already been deleted. 

These seem to indicate that implementing the feature would require something like `central user permission controller` that I did not expect to.

So at that time, I finally decided not to develop this feature 

but since you brought it up now, we can consider it a feature for next.




The simplest implementation is to allow 1Remote set a default account password. Once a user start a connection in MYSQL without setting account & password, the default account password is used to log in.

However, this solution only allows for setting the account password and I expect other configurations such as "script before connect" to also be localized.


Another solution is to keep a copy of the connection in MySQL in local sqlite, which I have described before. However, I haven't fully figured out the logic yet, but it's obvious that it requires a lot of development and there seem to be logical issues as well.

**Reply by majkinetor** [reply]
I have some ideas that could work and are trivial to implement. I will get back to this, I just added a note here as a placholder so I don't forget.

**Reply by majkinetor** [reply]
Created ticket #400 for this.
